packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.base.tsx

```diff
@@ -472,14 +472,13 @@ export class ContextualMenuBase extends React.Component<IContextualMenuProps, IC
     return (
       <ul className={this._classNames.list} onKeyDown={this._onKeyDown} onKeyUp={this._onKeyUp} role="menu">
         {menuListProps.items.map((item, index) => {
-          const menuItem = this._renderMenuItem({
-            ...item,
-            index,
-            focusableElementIndex: indexCorrection,
-            totalItemCount: menuListProps.totalItemCount,
-            hasCheckmarks: menuListProps.hasCheckmarks,
-            hasIcons: menuListProps.hasIcons,
-          });
+          let contextualMenuItemRenderProps = item as IContextualMenuItemRenderProps;
```

Comment:
I believe that this winds up mutating the items list itself. Part of the reason why we create a new object each time is to ensure that we don't change information that shouldn't be changed. 

Since it seems like this was a breaking perf change, I think we should consider changing the type of the `IContextualMenuItemRenderProps`. It could be that we go to something like `IContextualMenuItemRenderProps {
iitem: IContextualMenuItem}` or that get's removed entirely and the type of the render function changes to`defaultMenuItemRenderer: (item: IcontextualmenuItem, otherProps: {the other props}`

@dzearing since this caused such a big perf change is it alright to consider making a breaking change?

Reply:
@joschect @dzearing friendly ping on this. I was able to revisit this in a way that will allow us to still be backwards compatible. This regression is noticeable for perf and fixing this will increase subsequent persistent menu renders for web apps by ~10-15% going off of profiles I've been running with the change.